### DISCUSSION TITLE
FR(scm-diff-editor): Quit (q) should prompt to save (y / n / esc)

### DISCUSSION THREAD

**Discussion by avamsi** [arxanas/git-branchless#1037] [Ideas]
y would be Event::QuitAccept, n Event::QuitCancel and esc would cancel the prompt.

**Comment by arxanas** [comment]
What is the advantage of this workflow over the existing workflow?

**Reply by avamsi** [reply]
It would offer a chance to save dirty files (and so, not lose any edits unexpectedly).

I just realized hg diff editor (probably the closest thing to scm-diff-editor) doesn't do this -- somehow I don't think I ever ran into this with hg diff editor but I already quit scm-diff-editor multiple times by mistake (expecting a prompt -- maybe some muscle memory from https://github.com/zyedidia/micro?).

And in general, I'm hoping quit confirmation is not "out there" given it's quite common in editors?

**Reply by arxanas** [reply]
There should already be a confirmation dialog if you try to quit with nontrivial changes:

https://github.com/arxanas/git-branchless/blob/d898f86a38ddaeeff8c4434b18fa29adb1cac8a8/scm-record/tests/test_scm_record.rs#L311-L313

Is that what you're talking about, or something else?

(Ideally, we would simply save the editor state — it can already be serialized/deserialized — and resume it later; the quit dialog is the compromise given that resumption is not implemented.)

**Reply by avamsi** [reply]
ACK, this is what I'm talking about and I was just able to reproduce this too. Thanks!
(Now, I'm not really sure how I missed this earlier 🤔 -- sorry for the false alarm in any case.)
